File: src/main/java/hdzi/editstarters/ui/MavenButtonAction.java
package hdzi.editstarters.ui;

import com.intellij.openapi.actionSystem.DataContext;
import hdzi.editstarters.buildsystem.BuildSystem;
import hdzi.editstarters.buildsystem.maven.MavenBuildSystem;
import org.jetbrains.idea.maven.model.MavenConstants;

/**
 * maven项目Edit Starters按钮
 */
public class MavenButtonAction extends EditStartersButtonAction {

    @Override
    protected boolean isMatchFile(String name) {
        return MavenConstants.POM_XML.equals(name);
    }

    @Override
    protected BuildSystem newBuildSystem(DataContext dataContext) {
        return MavenBuildSystem.of(dataContext);
    }
}

File: src/main/kotlin/hdzi/editstarters/ui/MavenButtonAction.kt
package hdzi.editstarters.ui

import com.intellij.openapi.actionSystem.DataContext
import hdzi.editstarters.buildsystem.maven.MavenBuildSystem
import org.jetbrains.idea.maven.model.MavenConstants

/**
 * maven项目Edit Starters按钮
 */
class MavenButtonAction : EditStartersButtonAction() {
    override fun String.isMatchFile() = this == MavenConstants.POM_XML

    override fun newBuildSystem(dataContext: DataContext) = MavenBuildSystem(dataContext)

}